### DISCUSSION TITLE
Vue router. Call changeLocation twice, when use $router.push

### DISCUSSION THREAD

**Discussion by viiy** [quasarframework/quasar#14836] [General - Components / Directives / etc]
Quasar 2.10.1
vue-router 4.1.6

I have 3 routes
`const routes = [
  {
    path: "/",
    component: () => import("layouts/MainLayout.vue"),
    children: [
      {
        path: "users", component: () => import("pages/Users.vue"),
        name: 'users',
      },
      {
        path: "tables", component: () => import("pages/Tables.vue"),
        name: 'tables',
      },
      {
        path: "fields", component: () => import("pages/Fields.vue"),
        name: 'fields',
      }
}]`

When i use link (to="/users") all ok, internal vue-router function (changeLocation) calling one time.
But if i import `useRouter` and calling `$router.push('/users')` in component — changeLocation calling two times. 
Example, set brakepoint on ` changeLocation(to, state, replace)` in console, im on 'tables' page, calling `$router.push('/users')` **one time**, and in console i saw two breaks: 
` to="/tables', state={back: null, current: "/tables", forward: "/users": position: 1, replaced: true}, true`
` to="/users', state={back: "/tables", current: "/users", forward: null: position: 2, replaced: false}, false`

**Comment by MilosPaunovic** [comment]
This is an issue with `vue-router`, so I would suggest you post your question on their repository.
https://github.com/vuejs/router/issues
